gcovr/gcov_parser.py

```diff
@@ -231,6 +241,10 @@ def parse_metadata(lines: List[str]) -> Dict[str, str]:
         else:
             break  # stop at the first line that is not metadata
 
+    if "Source" not in collected:
+        data = "\n".join(lines)
+        raise RuntimeError(f"Missing key 'Source' in metadata. GCOV data was:\n{data}")
```

Comment:
Since the data is potentially corrupted, adding clear start/end markers to the error message could be helpful.